### PR TITLE
WebGui support for the illuminance sensors type.

### PR DESCRIPTION
**Pull request by GreenMarc**
Hi,
here are the pull requests for the sensor support. See the following forum threads.. : 

https://forum.pilight.org/Thread-BH1750-i2c-digital-illuminance-sensor
https://forum.pilight.org/Thread-TSL2561-i2c-digital-illuminance-sensor
https://forum.pilight.org/Thread-WebGUI-support-for-illuminance-sensors

thanks
Marc

### CODE REVIEW

**Comment by CurlyMoo** [comment]
1. Can adapt the PR according to the contribution guide **without closing this one**
2. Can adapt it so we get two PR's for each device **without closing this one**

**Comment by GreenMarc** [comment]
ok... i split the devices and the changes at the webgui... And i may read the contribution guide one more time... sry

Marc

**Comment by GreenMarc** [comment]
Ok, now another try.

I removed all changes for the 2 sensors. Only the changes for the webgui support are in this PR. So please take look at this changes.... : https://github.com/pilight/pilight/pull/308/files

The 2 sensors come in 2 seperate PRs in the next minutes....

Greetz
Marc

**Comment by CurlyMoo** [comment]
Can you make this compliant with the contribution guide.

File: libs/webgui/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-7.4
\ No newline at end of file
+7.4
```
Comment (pilino1234) [inline libs/webgui/version.txt]: Should be bumped to 7.5

File: libs/pilight/protocols/433.92/tcm.c [hunk 2]
```diff
@@ -93,9 +93,11 @@ static void parseCode(void) {
 	humidity = binToDecRev(binary, 16, 23);
 
 	temp = binToDecRev(binary, 24, 35);
+
 	if(binary[24]) {
 		temp -= 4096;
 	}
+
```
Comment (pilino1234) [inline libs/pilight/protocols/433.92/tcm.c]: Offtopic for this PR?

File: libs/webgui/pilight.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ var aTimers = new Array();
 var sDateTimeFormat = "HH:mm:ss YYYY-MM-DD";
 var aDateTimeFormats = new Array();
 var aWebcamUrl = new Array();
-var aDecimalTypes = ["temperature", "humidity", "wind", "pressure", "sunriseset"];
+var aDecimalTypes = ["temperature", "humidity", "wind", "pressure", "sunriseset","illuminance"];
```
Comment (pilino1234) [inline libs/webgui/pilight.js]: Can you add a space after the comma, like in the other lines?

File: libs/pilight/protocols/433.92/tcm.c (hunk 2 above)
Comment (GreenMarc) [inline libs/pilight/protocols/433.92/tcm.c]: yes, manual local conflicts edit...@ local merge

File: libs/webgui/version.txt (hunk 1 above)
Comment (GreenMarc) [inline libs/webgui/version.txt]: ok

File: libs/webgui/pilight.js (hunk 3 above)
Comment (GreenMarc) [inline libs/webgui/pilight.js]: ok

**Comment by pilino1234** [comment]
The changes look good 😄
Can you squash these commits into a single commit please? See the link (http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit) in the contributing.md for instructions on how to do this.

**Comment by GreenMarc** [comment]
argh....wth does the old tcm and generic dimmer versions in my patch.... ok, another try... sorry

**Comment by CurlyMoo** [comment]
Can you please just experiment on your own repo's / branches and fix this PR when you're done and figured everything out?

**But don't close this PR**.

**Comment by GreenMarc** [comment]
How I do it ? Every changes I do on my "support_illuminance_sensors" branch will automaticaly post on this PR ?!

**Comment by CurlyMoo** [comment]
You don't have to work on `support_illuminance_sensors` until you're done right? Just create a new branch called `experiment` or something not part of this PR.

**Comment by GreenMarc** [comment]
Since 5 Days, all i want to do is to understand this f*****g [b]git/github[/b] and post you a contribution-guide-conform patch against your act. development branch.

I works on an other -experimental-branch on the same thing. And there it is right -> https://github.com/GreenMarc/pilight/commits/webgui_copy

Now I try to repair this PR but with no success. I think the next step i do is to git reset all commits, git merge all your changes and start from new in this branch. This PR would be closed a couple of minutes befor i reopend it again... and i think the reviews & comments will be deleted ??!

**Comment by pilino1234** [comment]
You can use `git diff` to create a patch file based on the development branch (and backup the illuminance icon), `reset --hard` to the latest commit on the dev branch (BUT DON'T PUSH!), `git apply` the patch as a single commit, and then `push -f` to support_illuminance_sensors

**Comment by GreenMarc** [comment]
Hi, 
there are errors while applying the patch file . See the [logfile.txt](https://github.com/pilight/pilight/files/629523/logfile.txt)

Is this because the reset --hard has "patched" already the files ??

Here is the [reset.diff.txt](https://github.com/pilight/pilight/files/629528/reset.diff.txt)

Should I push my local branch with this errors ?

thanks...

**Comment by pilino1234** [comment]
You got your patch the wrong way around :P

**Comment by CurlyMoo** [comment]
>Since 5 Days, all i want to do is to understand this f*****g git/github

I encourage all learning tracks, but why on live and not on test environments?

**Comment by GreenMarc** [comment]
I startet with this web-based git here. It was all good, i found the fork function, i duplicated a branch, i edited my changes online here in the github webinterface, finaly i created a pull request. This was all easy and i must not know very much git knowledge to do this..... 
And then came you and the single commit.... Therese is no how-to to do this with this web-based github...... I also must try and error with the console git and many, many many Google lessons.... 
sry 
Marc

**Comment by wo-rasp** [comment]
Start reading here: https://forum.pilight.org/Thread-GIT-related-issues?pid=18552#pid18552

**Comment by CurlyMoo** [comment]
@GreenMarc, Can you post a screenshot of the webgui for the devices you use it with?

**Comment by GreenMarc** [comment]
screenshot_20170625-214048

**Comment by CurlyMoo** [comment]
Thanks!

**Comment by CurlyMoo** [comment]
Can you do rewrite as well?

**Pull request merged**